src/protocol/http/org/apache/jmeter/protocol/http/util/HTTPArgument.java

```diff
@@ -148,15 +149,19 @@ public HTTPArgument(String name, String value, boolean alreadyEncoded, String co
                 if(log.isDebugEnabled()) {
                     log.debug("Decoding name, calling URLDecoder.decode with '"+name+"' and contentEncoding:"+EncoderCache.URL_ARGUMENT_ENCODING);
                 }
-                name = URLDecoder.decode(name, EncoderCache.URL_ARGUMENT_ENCODING);
+                URLCodec urlCodec = new URLCodec(contentEncoding);
```

Comment:
In this code you use a different encoding than UTF-8. Are you sure it's allowed for parameter names ? If yes, then the PR should impact getEncodedName() which uses UTF-8 by default.